OrbitGl/GraphTrack.cpp

```diff
@@ -86,26 +72,73 @@ void GraphTrack::Draw(GlCanvas* canvas, PickingMode picking_mode) {
     if (it != values_.begin()) --it;
     uint64_t previous_time = it->first;
     double last_normalized_value = (it->second - min_) * inv_value_range_;
+    float kDotRadius = 2.5;
+    float base_y = pos_[1] - size_[1];
+    time_graph_->GetBatcher().AddCircle(
+        Vec2(time_graph_->GetWorldFromTick(previous_time),
+             base_y + static_cast<float>(last_normalized_value) * size_[1]),
+        kDotRadius, GlCanvas::kZValueText, kLineColor);
     for (++it; it != values_.end(); ++it) {
       if (previous_time > max_ns) break;
       uint64_t time = it->first;
       double normalized_value = (it->second - min_) * inv_value_range_;
-      float base_y = pos_[1] - size_[1];
       float x0 = time_graph_->GetWorldFromTick(previous_time);
       float x1 = time_graph_->GetWorldFromTick(time);
       float y0 = base_y + static_cast<float>(last_normalized_value) * size_[1];
       float y1 = base_y + static_cast<float>(normalized_value) * size_[1];
-      time_graph_->GetBatcher().AddLine(Vec2(x0, y0), Vec2(x1, y0), GlCanvas::kZValueText,
-                                        kLineColor);
-      time_graph_->GetBatcher().AddLine(Vec2(x1, y0), Vec2(x1, y1), GlCanvas::kZValueText,
+      // TODO: Draw only the dots when it isn't a continuous variable?
+      time_graph_->GetBatcher().AddLine(Vec2(x0, y0), Vec2(x1, y1), GlCanvas::kZValueText,
                                         kLineColor);
+      time_graph_->GetBatcher().AddCircle(Vec2(x1, y1), kDotRadius, GlCanvas::kZValueText,
+                                          kLineColor);
 
       previous_time = time;
       last_normalized_value = normalized_value;
     }
   }
 }
 
+void GraphTrack::DrawWhiteLabel(GlCanvas* canvas, double graph_value, uint64_t graph_time,
+                                float graph_y_pos) {
+  const TimeGraphLayout& layout = time_graph_->GetLayout();
+
+  // Draw label with graph value at current mouse position.
+  const Color kWhite(255, 255, 255, 255);
+  const Color kBlack(0, 0, 0, 255);
+
+  int text_width = canvas->GetTextRenderer().GetStringWidth(std::to_string(graph_value).c_str());
+  Vec2 white_text_box_size(text_width, layout.GetTextBoxHeight());
+  Vec2 value_in_screen(time_graph_->GetWorldFromTick(graph_time), graph_y_pos);
```

Comment:
value_in_screen is misleading, we are calling GetWorldFromTick for the x component, which makes me think that we are not in screen space but in world space. I would replace by "pos", or "target_pos" maybe?

Reply:
Sure!